File: BackEnd/Java/java-spring-maven-jpa-postgresql/src/main/java/com/pbshop/java/spring/maven/jpa/postgresql/opsdashboard/OpsDashboardController.java
package com.pbshop.java.spring.maven.jpa.postgresql.opsdashboard;

import java.util.Map;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.pbshop.java.spring.maven.jpa.postgresql.auth.security.AuthenticatedUserPrincipal;
import com.pbshop.java.spring.maven.jpa.postgresql.common.api.ApiResponse;

@RestController
@RequestMapping("${pbshop.api.base-path:/api/v1}/admin/ops-dashboard")
public class OpsDashboardController {

    private final OpsDashboardService opsDashboardService;

    public OpsDashboardController(OpsDashboardService opsDashboardService) {
        this.opsDashboardService = opsDashboardService;
    }

    @GetMapping("/summary")
    public ApiResponse<Map<String, Object>> summary(@AuthenticationPrincipal AuthenticatedUserPrincipal principal) {
        return ApiResponse.success(opsDashboardService.summary(principal));
    }
}

File: BackEnd/Kotlin/kotlin-ktor-gradle-exposeddao-postgresql/src/main/kotlin/com/pbshop/kotlin/ktor/gradle/exposeddao/postgresql/opsdashboard/OpsDashboardController.kt
package com.pbshop.kotlin.ktor.gradle.exposeddao.postgresql.opsdashboard

import com.pbshop.kotlin.ktor.gradle.exposeddao.postgresql.common.respondStub
import com.pbshop.kotlin.ktor.gradle.exposeddao.postgresql.security.PbRole
import com.pbshop.kotlin.ktor.gradle.exposeddao.postgresql.security.requireAnyRole
import io.ktor.server.application.call
import io.ktor.server.routing.Route
import io.ktor.server.routing.get

class OpsDashboardController(
    private val service: OpsDashboardService,
) {
    fun Route.register() {
        get("/admin/ops-dashboard/summary") {
            call.requireAnyRole(PbRole.ADMIN)
            call.respondStub(service.summary())
        }
    }
}
